Feature/ou-level-wizard #21

laurapareja opened the pull request:
### :pushpin: References

* **Issue:** Closes#21 Project Monitoring
Closes #89  

* **d2-ui-components**: Requires #89 

### :memo: Implementation
- Include `selectableLevels` prop in `OrgUnitTree` (by default is all selectable). 
- `selectableLevel` is an Array 
- the biggest number of `selectableLevel` hasn't children.
- Include `typeInput` prop in `OrgUnitTree` (by default is checkbox)

### :art: Screenshots
**Example with one `selectableLevels`**:
``` const selectableLevels = [3];```

![image](https://user-images.githubusercontent.com/24606534/68875037-74da1600-0702-11ea-84e7-736f9eddbafd.png)

**Example with `selectableLevels`**:
``` const selectableLevels = [1, 3];```

In this case, level 3 is the last selectable level so it doesn't have any children 

![image](https://user-images.githubusercontent.com/24606534/68952211-cb595a00-07bf-11ea-98a7-686911c6c310.png)

**Example without `selectableLevel` and `typeInput` props**, default OrgUnitTree

![image](https://user-images.githubusercontent.com/24606534/68952484-3b67e000-07c0-11ea-9cc0-fc60f21c009a.png)

laurapareja committed:
Change checkbox to radio button in Level 4

laurapareja committed:
Remove comments

laurapareja committed:
Fixed indentation

laurapareja committed:
Merge branch 'development' into feature/ou-level-wizard

laurapareja committed:
Change OrgUnitTree to be selectable only with prop selectableLevel

tokland commented on src/org-unit-tree/OrgUnitTree.component.js:
```diff
@@ -234,6 +244,14 @@ class OrgUnitTree extends React.Component {
 
+        const handleTypeLabel = () => {
+            if (selectableLevel !== undefined) {
+                return "radio";
```
I forgot about this: whether a radio or checkbox is to be used does not depend on `selectableLevel`, it should be a separate prop (if we really want to use a radio input here).

laurapareja replied:
In the last commit, you can see another prop `typeInput`. By default, without this prop, it would be checkbox.

laurapareja committed:
Implement styles selectableLevels + typeInput prop

laurapareja committed:
Change selectableLevel to selectableLevels

laurapareja committed:
Update README.md

Two more props for OrgUnitSelector:
- `typeInput`: `OrgUnitTree`/`TreeView` input can change with typeInputprop (by default is a checkbox)
- `selectableLevels`: it's an array of OU levels that can be selectable. The biggest number of selectableLevels has not children in OrgUnitTree
